dataclass_wizard/parsers.py

```diff
@@ -401,10 +401,7 @@ def __post_init__(self, cls: Type,
         # Total count should be `Infinity` here, since the variadic form
         # accepts any number of possible arguments.
         self.total_count: N = float('inf')
-        # Check for the count of parsers which don't handle `NoneType` - this
-        # should exclude the parsers for `Union` types that have `None` in the
-        # list of args.
-        self.required_count = 0 if None in self.first_elem_parser[0] else 1
```

Comment:
I don't actually remember what this line was actually doing 😅 

But I'm glad you caught it, that this is where we need to make the change. Totally right!